src/rpc/misc.cpp

```diff
@@ -37,6 +40,219 @@
+        c = (c << 4) + hexval(*p); // + takes precedence over <<
+        out.push_back(c);
+    }
+}
+
+void ProcessTxForHistory(const CTransaction &tx, const uint256 hashBlock, UniValue &entry, std::string walletAddress,
+                         int nHeight = 0, int nConfirmations = 0, int nBlockTime = 0)
+{
+    auto txid = tx.GetHash();
+    entry.pushKV("txid", txid.GetHex());
+    entry.pushKV("version", tx.nVersion);
+
+    if (tx.IsInvite()) {
+        entry.pushKV("isInvite", true);
+    }
+
+    UniValue vin(UniValue::VARR);
+    bool isSender = false;
+
+    if (tx.IsCoinBase()) {
+        entry.pushKV("isCoinbase", true);
+    } else {
+        for (unsigned int i = 0; i < tx.vin.size(); i++) {
+            const CTxIn& txin = tx.vin[i];
+            UniValue in(UniValue::VOBJ);
+            in.pushKV("txid", txin.prevout.hash.GetHex());
+            in.pushKV("vout", (int64_t)txin.prevout.n);
+
+            // Add address and value info if spentindex enabled
+            CSpentIndexValue spentInfo;
+            CSpentIndexKey spentKey(txin.prevout.hash, txin.prevout.n);
+            if (GetSpentIndex(spentKey, spentInfo)) {
+                if (tx.IsInvite()) {
+                    //invites are not demoninated in sotoshi.
+                    in.pushKV("amount", spentInfo.satoshis);
+                } else {
+                    in.pushKV("amount", ValueFromAmount(spentInfo.satoshis));
+                    in.pushKV("amountMicros", spentInfo.satoshis);
+                }
+
+                auto address = CMeritAddress{(signed char)(spentInfo.addressType), spentInfo.addressHash}.ToString();
+
+                in.pushKV("address", address);
+                isSender = isSender || address == walletAddress;
+
+                const auto maybe_referral = prefviewcache->GetReferral(spentInfo.addressHash);
+                if (maybe_referral) {
+                    in.pushKV("alias", maybe_referral->GetAlias());
+                }
+            } else {
+                debug("could not fetch spent info");
+            }
+
+            vin.push_back(in);
+        }
+
+        if (!isSender) {
+            auto firstInput = vin[0];
+            vin.setArray(); // Clear obj & set type to array
+            vin.push_back(firstInput); // Add only first input
+        }
+
+        entry.push_back(Pair("inputs", vin));
+    }
+
+    int64_t totalAmount = 0;
+    UniValue vout(UniValue::VARR);
+    bool couldBeGrowthReward = false;
+    bool isMarket = false;
+    bool isPoolReward = false;
+
+    for (unsigned int i = 0; i < tx.vout.size(); i++) {
+        const auto& txout = tx.vout[i];
+        UniValue out(UniValue::VOBJ);
+        txnouttype type;
+        std::vector<CTxDestination> addresses;
+        int required;
+        bool isChangeOutput;
+
+        if (ExtractDestinations(txout.scriptPubKey, type, addresses, required)) {
+            const CTxDestination dest = addresses[0];
```

Comment:
const auto

Reply:
Also is it possible for addresses here to be empty? if not add an assert(!addresses.empty())